documentation/docs/01-routing.md

```diff
@@ -0,0 +1,136 @@
+
+```html
+<!-- src/routes/about.svelte -->
+<svelte:head>
+	<title>About</title>
+</svelte:head>
+
+<h1>About this site</h1>
+<p>TODO...</p>
+```
+
+Dynamic parameters are encoded using `[brackets]`. For example, a blog post might be defined by `src/routes/blog/[slug].svelte`. Soon, we'll see how to access that parameter in a [load function](#Loading) or the [page store](#TODO).
+
+### Endpoints
+
+Endpoints are modules written in `.js` (or `.ts`) files that export functions corresponding to HTTP methods. Each function receives HTTP `request` and `context` objects as arguments. This is useful for creating a JSON API. For example our hypothetical blog page, `/blog/cool-article`, might request data from `/blog/cool-article.json`, which could be represented by a `src/routes/blog/[slug].json.js` endpoint:
+
+```js
+import db from '$lib/database';
+
+/**
+ * @param {{
+ *   host: string;
+ *   method: 'GET';
+ *   headers: Record<string, string>;
+ *   path: string;
+ *   body: any;
+ *   query: URLSearchParams;
+ * }} request
+ * @param {any} context
+ * @returns {{
+ *   status?: number;
+ *   headers?: Record<string, string>;
+ *   body: any;
+ * }}
+ */
+export async function get(request, context) {
+	// the `slug` parameter is available because this file
+	// is called [slug].json.js
+	const { slug } = request.params;
+
+	const article = await db.get(slug);
+
+	if (article !== null) {
+		return {
+			body: {
+				article
+			}
+		};
+	} else {
+		return {
+			status: 404,
+			body: {
+				error: 'Not found'
+			}
+		};
+	}
+}
+```
+
+Because this module only runs on the server (or when you build your site, if [prerendering](#Prerendering)), you can freely access things like databases. (Don't worry about `$lib`, we'll get to that [later](#$lib).)
+
+The second argument, `context`, is something you define during [setup](#Setup), if necessary.
+
+The job of this function is to return a `{status, headers, body}` object representing the response. If the returned `body` is an object, and no `content-type` header is returned, it will automatically be turned into a JSON response.
+
+> We don't interact with the `req`/`res` objects you might be familiar with from Node's `http` module or frameworks like Express, because they're only available on certain platforms. Instead, SvelteKit translates the returned object into whatever's required by the platform you're deploying your app to.
+
+For endpoints that handle other HTTP methods, like POST, export the corresponding function:
+
+```js
+export function post(request, context) {...}
+```
+
+Since `delete` is a reserved word in JavaScript, DELETE requests are handled with a `del` function.
+
+
+### Private modules
+
+A filename that has a segment with a leading underscore, such as `src/routes/foo/_Private.svelte` or `src/routes/bar/_utils/cool-util.js`, is hidden from the router, but can be imported by files that are not.
```

Comment:
Is there any reason you'd do this instead of just putting it outside `routes`?

Reply:
`$lib` is for stuff that's used in multiple places. it's not uncommon to have components or modules that are only useful for a specific route